src/app/project-sets/[projectSetId]/page.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react"
+import {Text} from "@/components/ui/text"
+import {type Project, ProjectRequirement, ProjectSet} from "@/_temp_types/projects"
+import {Button} from "@/components/ui/button"
+import {DataTable} from "@/components/ui/data-table"
+import {mutableColumnDefs, persistedColumnDefs} from "@/app/project-sets/[projectSetId]/columns"
+import {redirect} from "next/navigation"
+import {ApiTeamSetTemplate} from "@/_temp_types/api/teams"
+import {toast} from "@/components/ui/use-toast"
+import {ProjectSetSelect} from "@/app/project-sets/[projectSetId]/(components)/projectSetSelect"
+import {SidebarProjectsDisplay} from "@/app/project-sets/[projectSetId]/(components)/sidebarProjectsDisplay"
+import {EditModeButton} from "@/app/project-sets/[projectSetId]/(components)/editModeButton"
+import {NumProjectsSubtitle} from "@/app/project-sets/[projectSetId]/(components)/numProjectsSubtitle"
+
+async function getRawProjectSetsData(): Promise<ApiTeamSetTemplate[]> {
```

Comment:
We do not need to fetch the full data for every project set in the course just for the dropdown.

We should fetch just the name + id of every project set in the course (this is for the dropdown)
When you load the page, you fetch the detailed data for a specific project set (this is for the rest of the page)